auth_backends/tests/test_pipeline.py

```diff
@@ -11,33 +11,128 @@
 
 
 class GetUserIfExistsPipelineTests(TestCase):
-    """ Tests for the get_user_if_exists pipeline function. """
+    """
+    Tests for the get_user_if_exists pipeline function.
+    """
 
     def setUp(self):
         super().setUp()
         self.username = 'edx'
         self.details = {'username': self.username}
 
-    def test_no_user_exists(self):
-        """ Verify an empty dict is returned if no user exists. """
+    @patch('auth_backends.pipeline.set_custom_attribute')
+    @patch('auth_backends.pipeline.IGNORE_LOGGED_IN_USER_ON_MISMATCH.is_enabled')
+    def test_no_user_exists(self, mock_toggle, mock_set_attribute):
+        """
+        Verify an empty dict is returned if no user exists.
+        """
+        mock_toggle.return_value = True
+
         actual = get_user_if_exists(None, self.details)
         self.assertDictEqual(actual, {})
 
-    def test_existing_user(self):
-        """ Verify a dict with the user and extra details is returned if the user exists. """
+        mock_set_attribute.assert_not_called()
+
+    @patch('auth_backends.pipeline.set_custom_attribute')
+    @patch('auth_backends.pipeline.IGNORE_LOGGED_IN_USER_ON_MISMATCH.is_enabled')
+    def test_existing_user(self, mock_toggle, mock_set_attribute):
+        """
+        Verify a dict with the user and extra details is returned if the user exists.
+        """
+        mock_toggle.return_value = True
         user = User.objects.create(username=self.username)
+
         actual = get_user_if_exists(None, self.details)
         self.assertDictEqual(actual, {'is_new': False, 'user': user})
 
-    def test_get_user_if_exists(self):
-        """ Verify only the details are returned if a user is passed to the function. """
+        mock_set_attribute.assert_not_called()
+
+    @patch('auth_backends.pipeline.set_custom_attribute')
+    @patch('auth_backends.pipeline.IGNORE_LOGGED_IN_USER_ON_MISMATCH.is_enabled')
+    def test_get_user_if_exists_with_user_provided_no_mismatch(self, mock_toggle, mock_set_attribute):
+        """
+        Verify only the details are returned if a user is passed with no username mismatch.
+        """
+        mock_toggle.return_value = True
         user = User.objects.create(username=self.username)
+
         actual = get_user_if_exists(None, self.details, user=user)
         self.assertDictEqual(actual, {'is_new': False})
 
+        mock_set_attribute.assert_any_call('get_user_if_exists.username_mismatch', False)
+        mock_set_attribute.assert_any_call('get_user_if_exists.ignore_toggle_enabled', True)
+
+    @patch('auth_backends.pipeline.logger')
+    @patch('auth_backends.pipeline.set_custom_attribute')
+    @patch('auth_backends.pipeline.IGNORE_LOGGED_IN_USER_ON_MISMATCH.is_enabled')
+    def test_username_mismatch_toggle_enabled(self, mock_toggle, mock_set_attribute, mock_logger):
```

Comment:
It's ok to leave this as a separate test, since you expect something different. That said, you could still use `@ddt` and just have slightly different assertions for the two cases.